library/ccm.c

```diff
@@ -49,12 +49,14 @@
 #endif /* MBEDTLS_PLATFORM_C */
 #endif /* MBEDTLS_SELF_TEST && MBEDTLS_AES_C */
 
-#if !defined(MBEDTLS_CCM_ALT)
-
+#if !defined(MBEDTLS_CCM_ALT) || defined(MBEDTLS_SELF_TEST)
 /* Implementation that should never be optimized out by the compiler */
 static void mbedtls_zeroize( void *v, size_t n ) {
     volatile unsigned char *p = (unsigned char*)v; while( n-- ) *p++ = 0;
 }
+#endif
+
+#if !defined(MBEDTLS_CCM_ALT)
```

Comment:
Why did we need to change these ifdefs? If it was only because we were using `zeroize()` in the self-test, I don't think we need to change the ifdefs anymore because we shouldn't need to use `zeroize()` in the self-test.

Reply:
yes, I believe it was because of that reason. I'll revert

Reply:
Anyway, once merged, `mbedtls_platform_zeroize()` should be used, and this part will be removed